lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -370,4 +378,24 @@ public function prepare($statement)
 
         return parent::prepare($statement);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ping()
+    {
+        $this->connect('master');
```

Comment:
Seems incorrect to me: it should simply ping whichever connection is already selected, as otherwise we may attempt to select the slave and then not get a valid response

Reply:
will revert